### PR TITLE
Refactor ipam.go

### PR DESCRIPTION
**Pull request by iiiceoo**
- Refactor the code structure of the Allocate function.
- Fixed the issue that StatefulSet static IP prompt is missing when the interface name was incorrectly specified in the multi-NIC case.

Signed-off-by: iiiceoo <ziqian.xue@daocloud.io>

### CODE REVIEW

File: pkg/ipam/ipam.go [hunk 1]
```diff
@@ -77,68 +80,128 @@ func (i *ipam) Allocate(ctx context.Context, addArgs *models.IpamAddArgs) (*mode
...
+
+	return i.allocateInStandardMode(ctx, addArgs, pod, endpoint)
+}
 
-		if ipamAddResp != nil {
-			return ipamAddResp, nil
+func (i *ipam) retrieveStsIPAllocation(ctx context.Context, containerID, nic string, pod *corev1.Pod, endpoint *spiderpoolv1.SpiderEndpoint) (*models.IpamAddResponse, error) {
+	if !i.ipamConfig.EnabledStatefulSet || podmanager.GetControllerOwnerType(pod) != constant.OwnerStatefulSet {
+		return nil, nil
+	}
```
Comment (Icarus9913) [inline pkg/ipam/ipam.go:117]: 开关判断放到ipam层，非内部

File: pkg/ipam/ipam.go [hunk 2]
```diff
@@ -77,68 +80,128 @@ func (i *ipam) Allocate(ctx context.Context, addArgs *models.IpamAddArgs) (*mode
...
+	if endpoint == nil || endpoint.Status.Current == nil {
+		return nil, nil
+	}
```
Comment (Icarus9913) [inline pkg/ipam/ipam.go:120]: https://github.com/spidernet-io/spiderpool/blob/522a84cc580763d54a56cb3bfeae3d870b166ddd/pkg/ipam/ipam_statefulset.go#L22-L30
建议原处理方式

File: pkg/gcmanager/pod_cache.go [hunk 3]
```diff
@@ -138,7 +138,7 @@ func (s *SpiderGC) buildPodEntry(oldPod, currentPod *corev1.Pod, deleted bool) (
 	// deleted pod
 	if deleted {
 		// check StatefulSet pod, we will trace it if its controller StatefulSet object was deleted or decreased its replicas and the pod index was out of the replicas.
-		if s.gcConfig.EnableStatefulSet && podmanager.GetControllerOwnerType(currentPod) == constant.OwnerStatefulSet {
+		if s.gcConfig.EnableStatefulSet {
```
Comment (weizhoublue) [inline pkg/gcmanager/pod_cache.go:141]: ？

File: pkg/gcmanager/scanAll_IPPool.go [hunk 4]
```diff
@@ -92,7 +91,7 @@ func (s *SpiderGC) executeScanAll(ctx context.Context) {
 				// case: The pod in IPPool's ip-allocationDetail is not exist in k8s
 				if apierrors.IsNotFound(err) {
 					// check StatefulSet pod whether need to clean up its IP and Endpoint or not
-					if s.gcConfig.EnableStatefulSet && poolIPAllocation.OwnerControllerType == constant.OwnerStatefulSet {
+					if s.gcConfig.EnableStatefulSet {
```
Comment (weizhoublue) [inline pkg/gcmanager/scanAll_IPPool.go:94]: ？

**Review by weizhoublue — DISMISSED** [review]

**Review by Icarus9913 — APPROVED** [review]

**Review by weizhoublue — APPROVED** [review]

**Pull request merged**